testing_and_setup/compass/ocean/global_ocean/ARM60to10/init/define_base_mesh.py

```diff
@@ -1,41 +1,123 @@
+       lat : numpy.ndarray
+            latitude, vector of length m, with entries between -90 and 90,
+            degrees
+       lon : numpy.ndarray
+            longitude, vector of length n, with entries between -180 and 180,
+            degrees
+    """
+    # To speed up for testing, set following line to 1.0 degrees
+    dlon = 0.1
+    dlat = dlon
+    nlon = int(360. / dlon) + 1
+    nlat = int(180. / dlat) + 1
+    lon = np.linspace(-180., 180., nlon)
+    lat = np.linspace(-90., 90., nlat)
 
+    # Create cell width vs latitude for Atlantic and Pacific basins
     QU1 = np.ones(lat.size)
     EC60to30 = mdt.EC_CellWidthVsLat(lat)
     RRS30to10 = mdt.RRS_CellWidthVsLat(lat, 30, 10)
-
     AtlNH = RRS30to10
-    AtlGrid = mdt.mergeCellWidthVsLat(lat, EC60to30, AtlNH, 0, 6)
-
+    AtlVsLat = mdt.mergeCellWidthVsLat(lat, EC60to30, AtlNH, 0, 6)
     PacNH = mdt.mergeCellWidthVsLat(lat, 30 * QU1, RRS30to10, 50, 10)
-    PacGrid = mdt.mergeCellWidthVsLat(lat, EC60to30, PacNH, 0, 6)
+    PacVsLat = mdt.mergeCellWidthVsLat(lat, EC60to30, PacNH, 0, 6)
+
+    # Expand from 1D to 2D
+    _, AtlGrid = np.meshgrid(lon, AtlVsLat)
+    _, PacGrid = np.meshgrid(lon, PacVsLat)
+
+    # Signed distance of Atlantic region
+    fc = read_feature_collection('Atlantic_region.geojson')
+    signed_distance = signed_distance_from_geojson(fc, lon, lat,
+                                                   max_length=0.25)
+
+    # Merge Atlantic and Pacific distrubutions smoothly
+    transitionWidth = 500.0e3  # [m]
+    maskSmooth = 0.5 * (1 + np.tanh(signed_distance / transitionWidth))
+    cellWidthSmooth = AtlGrid * maskSmooth + PacGrid * (1 - maskSmooth)
+
+    # Merge Atlantic and Pacific distrubutions with step function
+    maskStep = 0.5 * (1 + np.sign(signed_distance))
+    cellWidthStep = AtlGrid * maskStep + PacGrid * (1 - maskStep)
 
-    cellWidth = mdt.AtlanticPacificGrid(lat, lon, AtlGrid, PacGrid)
+    # Create a land mask that is 1 over land
+    fc = read_feature_collection('Americas_land_mask.geojson')
+    Americas_land_mask = mask_from_geojson(fc, lon, lat)
+    fc = read_feature_collection('Europe_Africa_land_mask.geojson')
+    Europe_Africa_land_mask = mask_from_geojson(fc, lon, lat)
+    land_mask = np.fmax(Americas_land_mask, Europe_Africa_land_mask)
+
+    # Merge: step transition over land, smooth transition over water
+    cellWidth = cellWidthStep * land_mask + cellWidthSmooth * (1 - land_mask)
+
+    # save signed distance to a file
+    # da = xarray.DataArray(signed_distance,
+    #                      dims=['y', 'x'],
+    #                      coords={'y': lat, 'x': lon},
+    #                      name='signed_distance')
+    #cw_filename = 'signed_distance.nc'
+    # da.to_netcdf(cw_filename)
+
+# replace this:
+    #cellWidth = mdt.AtlanticPacificGrid(lat, lon, AtlGrid, PacGrid)
 
     import matplotlib.pyplot as plt
     import matplotlib
     matplotlib.use('Agg')
```

Comment:
Not directly related to the PR, but this should get moved to the top and should be reordered:
```
    import matplotlib
    matplotlib.use('Agg')
    import matplotlib.pyplot as plt
```
The switch to the `Agg` backend should be before importing `plt`.